flask_multipass/providers/ldap/providers.py

```diff
@@ -232,3 +232,21 @@ def search_groups(self, name, exact=False):
             for group_dn, group_data in self._search_groups(search_filter):
                 group_name = to_unicode(group_data[self.ldap_settings['gid']][0])
                 yield self.group_class(self, group_name, group_dn)
+
+
+class AuthFallbackLDAPIdentityProvider(LDAPIdentityProvider):
+    """Provides identity information using LDAP with a fallback to authorization data.
+
+    This identity provider is meant to be used together with the auth providers, like
+    Shibboleth, who can authorize access to users who are not present is the LDAP directory.
+    By default it will use the identifier returned by the auth provider and lookup all the data 
+    in LDAP.
+
+    In case the user does not have data in LDAP, the data provided via auth provider will be used.
+    """
```

Comment:
```suggestion
    """Provides identity information using LDAP with a fallback to auth provider data.
    
    This identity provider is meant to be used together with an auth provider that provides
    all the required data (in particular the Shibboleth provider).
    
    By default it will use only the identifier from the auth provider and look up all the data
    from LDAP.
    
    In case the user does not have data in LDAP however, the data provided from the auth provider
    will be used.
    """
```

I think my original text is clearer, and there was a typo in yours ;)